### PR TITLE
[IMP] stock: revamp

### PR DESCRIPTION
**Pull request by mash-odoo**
**In this commit**

- The detailed_type 'storable product' is removed and instead we add a Boolean field to track inventory.
- The tracking field should be visible only after the boolean field to track inventory is enabled

**Task**: 3526304

### CODE REVIEW

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo r+

**Comment by pedrobaeza** [comment]
Hi, can you please explain the reasoning behind changing this established method of setting the type to switch to a boolean that provokes that developers should check 2 fields for most of the operations? This is also for having the arguments to explain to our users.

And if I understand well, you can have products with type='event' for example, but with the check `is_storable`.

**Comment by ossimantylahti** [comment]
I'm with @pedrobaeza on this. This will break backward compatibility. Why is this being done? Are there any benefits that justify this change?

**Comment by legalsylvain** [comment]
Hi. Just tested on a demo database.

IMO, The text is quite confusing : both helper mention of consumable products, but not with the same consequency. (allways available vs to be inventoried).

image

image


> And if I understand well, you can have products with type='event' for example, but with the check is_storable.

Not via UI, because the "track inventory" checkbox is hidden for services. Indeed via ORM, there is no constrains. @mash-odoo : could you add one ? 

thanks !

**Comment by andrewdhastings** [comment]
I will agree with others, What is the benefit of this?

**Comment by desdelinux** [comment]
Can you explain the reason for this change? I can't see what benefit it brings... In addition, this will have an impact on all third-party developments

**Comment by janbec** [comment]
In our company of 80 users, I can honestly say that the difference between consumable and storeable has been easy to explain and has never been the source of issues/confusion. I also think that this breaks more (3rd party) than it helps.

**Comment by legalsylvain** [comment]
from my point of view (I work in a 300-person company where many people are not very comfortable with computers), I find that the new design is more explicit than the old one, and that asking the question “To be inventoried” (Yes / No) explicitly will be less of a source of error.

**Comment by amoyaux** [comment]
Hello everyone. The purpose of this pull request is to mainly remove the confusion between consumable and product type for new users. This pr is in a batch of multiple ones and the consumable type will become goods instead of consumable.
We want to keep a model similar to reality, that's why we move the storable part in a new field even if I understand it could be annoying to migrate your module due to that. The storable should only be able for consumable type not the other. The purpose of the compute is to remove storable when a type different from consumable is selected and I will probably add a constraint on that. Some UX changes still coming too

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo r+

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -725,12 +722,16 @@ class ProductTemplate(models.Model):
     show_forecasted_qty_status_button = fields.Boolean(compute='_compute_show_qty_status_button')
 
     @api.depends('type')
+    def compute_is_storable(self):
+        self.filtered(lambda t: t.type != 'consu' and t.is_storable).is_storable = False
```
[the remark this answers is not in the document]
Comment (ticodoo) [reply addons/stock/models/product.py:726]: Because if the product is already `is_storable = False`, there's no need to try to assign it again to `False`

**Pull request closed without merging**